crates/meilisearch/src/routes/chats/settings.rs

```diff
@@ -0,0 +1,262 @@
+use actix_web::web::{self, Data};
+use actix_web::HttpResponse;
+use deserr::Deserr;
+use index_scheduler::IndexScheduler;
+use meilisearch_types::deserr::DeserrJsonError;
+use meilisearch_types::error::deserr_codes::*;
+use meilisearch_types::error::{Code, ResponseError};
+use meilisearch_types::features::{
+    ChatCompletionPrompts as DbChatCompletionPrompts, ChatCompletionSettings,
+    ChatCompletionSource as DbChatCompletionSource, DEFAULT_CHAT_SEARCH_DESCRIPTION_PROMPT,
+    DEFAULT_CHAT_SEARCH_INDEX_UID_PARAM_PROMPT, DEFAULT_CHAT_SEARCH_Q_PARAM_PROMPT,
+    DEFAULT_CHAT_SYSTEM_PROMPT,
+};
+use meilisearch_types::keys::actions;
+use meilisearch_types::milli::update::Setting;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use super::ChatsParam;
+use crate::extractors::authentication::policies::ActionPolicy;
+use crate::extractors::authentication::GuardedData;
+use crate::extractors::sequential_extractor::SeqHandler;
+
+pub fn configure(cfg: &mut web::ServiceConfig) {
+    cfg.service(
+        web::resource("")
+            .route(web::get().to(SeqHandler(get_settings)))
+            .route(web::patch().to(SeqHandler(patch_settings)))
+            .route(web::delete().to(SeqHandler(reset_settings))),
+    );
+}
+
+async fn get_settings(
+    index_scheduler: GuardedData<
+        ActionPolicy<{ actions::CHATS_SETTINGS_GET }>,
+        Data<IndexScheduler>,
+    >,
+    chats_param: web::Path<ChatsParam>,
+) -> Result<HttpResponse, ResponseError> {
+    index_scheduler.features().check_chat_completions("using the /chats/settings route")?;
+
+    let ChatsParam { workspace_uid } = chats_param.into_inner();
+
+    // TODO do a spawn_blocking here ???
```

Comment:
I don't think that is necessary: this operation is not expected to take a long time in general, unlike, say, a search.